tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -560,27 +702,50 @@ TILEDB_EXPORT int32_t tiledb_filestore_size(
     return TILEDB_ERR;
   }
 
-  Context context(ctx, false);
-  Array array(context, std::string(filestore_array_uri), TILEDB_READ);
+  tiledb::sm::Context& context = ctx->context();
+  tiledb::sm::Array array(
+      tiledb::sm::URI(filestore_array_uri), context.storage_manager());
+
+  throw_if_not_ok(array.open(
+      tiledb::sm::QueryType::READ,
+      tiledb::sm::EncryptionType::NO_ENCRYPTION,
+      nullptr,
+      0));
+
+  tiledb::sm::Datatype dtype;
+  bool has_key;
+  throw_if_not_ok(array.has_metadata_key(
+      tiledb::sm::constants::filestore_metadata_size_key.c_str(),
+      &dtype,
+      &has_key));
 
-  tiledb_datatype_t dtype;
-  if (!array.has_metadata(
-          tiledb::sm::constants::filestore_metadata_size_key, &dtype)) {
+  if (!has_key) {
     LOG_STATUS_NO_RETURN_VALUE(Status_Error(
         std::string("Filestore size key not found in array metadata; this "
                     "filestore may not have been imported: ") +
         filestore_array_uri));
     return TILEDB_ERR;
   }
   uint32_t num;
-  const void* file_size;
-  array.get_metadata(
-      tiledb::sm::constants::filestore_metadata_size_key,
+  const void* file_size = nullptr;
+  throw_if_not_ok(array.get_metadata(
+      tiledb::sm::constants::filestore_metadata_size_key.c_str(),
       &dtype,
       &num,
-      &file_size);
+      &file_size));
+
+  if (!file_size) {
+    auto st = Status_Error(
+        "The array metadata doesn't contain the " +
```

Comment:
The existence of the metadata key is already checked above, no need to do it again here.

Reply:
but we should still null check the file_size attribute. I could throw a std::logic_error here and this is what I will do for now.

Reply:
We checked for the key existence prior to getting file_size. Any internal error in metadata that would render file_size=nullptr should be caught in the previous `throw_if_not_ok(array.get_metadata...`. I would say it's safe to assume file_size != nullptr here at this point.